pkg/webhook/webhook.go

```diff
@@ -259,6 +259,15 @@ func writeResponse(w http.ResponseWriter, ar *v1beta1.AdmissionReview) {
 	w.Write(resp)
 }
 
+func patchEmptyResources(patch []jsonPatchOperation, containerIndex uint, key string) []jsonPatchOperation {
```

Comment:
This looks clean and tidy, perhaps we could follow the same way and abstract `resources` patching in a separate function as well. This can be done via further PRs.